test_fms/mpp/test_mpp_mem_dump.F90

```diff
@@ -32,13 +32,10 @@ program test_mpp_mem_dump
   implicit none
 
   real :: memuse
-  integer(kind=INT_KIND) :: return_val = 0
 
   call mpp_mem_dump(memuse)
 
   if (memuse .LT. 0.0) then
-    return_val = 1
+    stop "mpp_mem_dump return negative value for memory"
   endif
-
-  stop return_val
 end program test_mpp_mem_dump
```

Comment:
If you want to return an integer, why not just wrap `stop` statement in the if block
```fortran
if (memuse .LT. 0.0) then		￼  
￼    stop 1		￼    
￼endif
```
Or use parameter integers to do the same thing and always stop with a value
```fortran
integer, parameter :: return_error = 1
integer, parameter :: return_success = 0
...
if (memuse .LT. 0.0) then
￼    stop return_error
else
    stop return_success
endif
```

Reply:
The value of the return is not important, other than it _must_ be non-zero.  Adding the character constant will have `STOP` end with a non-zero, I just don't get to choose the value of the exit.